File: app/src/main/java/com/example/e_trash/helper/ConfiguracaoFirebase.java
package com.example.e_trash.helper;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.storage.*;

public class ConfiguracaoFirebase {

    private static DatabaseReference referenciaFirebase;
    private static FirebaseAuth referenciaAutenticacao;
    private static StorageReference storage;


    //retorna a referencia do database
    public static DatabaseReference getFirebase(){
        if( referenciaFirebase == null ){
            referenciaFirebase = FirebaseDatabase.getInstance().getReference();
        }
        return referenciaFirebase;
    }

    // Retorna a instancia do Firebase
    public static FirebaseAuth getReferenciaAutenticacao() {
        if (referenciaAutenticacao == null) {
            referenciaAutenticacao = FirebaseAuth.getInstance();
        }
        return referenciaAutenticacao;
    }
    public static StorageReference getFirebaseStorage(){
        if( storage == null ){
            storage = FirebaseStorage.getInstance().getReference();
        }
        return storage;
    }



}

File: app/src/main/java/com/example/e_trash/helper/ConfiguracaoFirebase.kt
package com.example.e_trash.helper

import com.google.firebase.auth.FirebaseAuth

class ConfiguracaoFirebase {
	private lateinit var referenciaAutenticacao: FirebaseAuth

	fun getFirebaseAutenticacao(): FirebaseAuth {
		if (referenciaAutenticacao == null) {
			referenciaAutenticacao = FirebaseAuth.getInstance()
		}
		return referenciaAutenticacao
	}
}